### ISSUE TITLE
Lodestar forwards invalid attestation from gossip

### ISSUE THREAD

**Issue by philknows** [ChainSafe/lodestar#4412]
**Describe the bug**

Lodestar has been observed as attesting to bad blocks.

Reported from Nishant at Prysm on Eth R&D #interop channel. 

```
DEBUG sync: Gossip message was rejected agent=js-libp2p/0.36.2 error=bad block referenced in attestation data gossip score=0 multiaddress=/ip4/x.x.x.x/tcp/x peer id=xxx topic=/eth2/c2ce3aa8/beacon_aggregate_and_proof/ssz_snappy
DEBUG sync: Gossip message was rejected agent=js-libp2p/0.36.2 error=bad block referenced in attestation data gossip score=0 multiaddress=/ip4/x.x.x.x/tcp/x  peer id=xxx topic=/eth2/c2ce3aa8/beacon_aggregate_and_proof/ssz_snappy
DEBUG sync: Gossip message was rejected agent=js-libp2p/0.36.2 error=bad block referenced in attestation data gossip score=0 multiaddress=/ip4/x.x.x.x/tcp/x  peer id=xxx topic=/eth2/c2ce3aa8/beacon_aggregate_and_proof/ssz_snappy
```

Block in question: https://goerli.beaconcha.in/block/3665805

```
DEBUG sync: Received block blockSlot=3665805 graffiti=Nimbus/v22.7.0-9e6d9b-stateofus proposerIndex=191467 sinceSlotStartTime=2.893640219s
DEBUG powchain: Latest eth1 chain event blockHash=0x96c3290f7e79a866a134a583dcb6dd577eb915fb36c5e979271d4ae2c50c44da blockNumber=7404349 difficulty=0
DEBUG sync: Inserting in invalid block cache root=0x608e70b04b3912d8731efa17326004b444d30430e77f9a32acdf229404ce8c93
ERROR sync: Could not handle p2p pubsub error=could not process block: could not validate new payload: received an INVALID_BLOCK_HASH payload from execution engine topic=/eth2/c2ce3aa8/beacon_block/ssz_snappy
```

**Expected behavior**

Should not be attesting to bad blocks. Potentially related to https://github.com/ChainSafe/lodestar/issues/4388 ?

**Comment by dapplion** [comment]
Note it's not clear if a Lodestar node produced this attestation, the one that forwarded the attestation gossip message does any validators connected to it.

**Comment by twoeths** [comment]
lodestar does catch the invalid block

```
Aug-14 17:21:00.665[NETWORK]       ^[[36mverbose^[[39m: Received gossip block slot=3665805, root=0x608e…8c93, curentSlot=3665805, peerId=bafzaajiiaijccav37zyt7s5jgx2vkdtuclgnagytf6ci2s7a6zz3iufzac5q24iv3a, delaySec=0.6570000648498535
Aug-14 17:21:02.310[CHAIN]           ^[[31merror^[[39m: Block error slot=3665805, errCode=BLOCK_ERROR_EXECUTION_ERROR BLOCK_ERROR_EXECUTION_ERROR
Error: BLOCK_ERROR_EXECUTION_ERROR
    at verifyBlockExecutionPayload (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:298:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at verifyBlocksExecutionPayload (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:124:31)
    at async Promise.all (index 2)
    at BeaconChain.verifyBlocksInEpoch (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlock.ts:67:91)
    at BeaconChain.processBlocks (file:///usr/app/packages/beacon-node/src/chain/blocks/index.ts:71:68)
    at Timeout.JobItemQueue.runJob [as _onTimeout] (file:///usr/app/packages/beacon-node/src/util/queue/itemQueue.ts:92:22)
Aug-14 17:21:02.311[NETWORK]         ^[[31merror^[[39m: Error receiving block slot=3665805, peer=bafzaajiiaijccav37zyt7s5jgx2vkdtuclgnagytf6ci2s7a6zz3iufzac5q24iv3a code=BLOCK_ERROR_EXECUTION_ERROR, execStatus=INVALID_BLOCK_HASH, errorMessage=Malformed block
Error: BLOCK_ERROR_EXECUTION_ERROR
    at verifyBlockExecutionPayload (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:298:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at verifyBlocksExecutionPayload (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:124:31)
    at async Promise.all (index 2)
    at BeaconChain.verifyBlocksInEpoch (file:///usr/app/packages/beacon-node/src/chain/blocks/verifyBlock.ts:67:91)
    at BeaconChain.processBlocks (file:///usr/app/packages/beacon-node/src/chain/blocks/index.ts:71:68)
    at Timeout.JobItemQueue.runJob [as _onTimeout] (file:///usr/app/packages/beacon-node/src/util/queue/itemQueue.ts:92:22)
```
it means the `beacon_aggregate_and_proof` message should be IGNORED since the head block is not in forkchoice

I think the message was sent to peers through control mechanism (IHAVE/IWANT), there was a bug that libp2p-gossipsub sent not validated messages https://github.com/ChainSafe/js-libp2p-gossipsub/pull/277

all of lodestar nodes use libp2p-gossipsub 0.14.0 so it does not have the fix

**Comment by philknows** [comment]
As discussed, we will release `libp2p-gossipsub 0.14.1` with a hotfix to resolve first.

**Comment by twoeths** [comment]
reopening the issue as this could cause some memory issues, also note that this is resolved if we migrate to esm version of libp2p

cc @philknows @wemeetagain

**Comment by philknows** [comment]
Hey @tuyennhv, since we confirmed this fix is not causing the memory leak on `v1.1.0` can we close again?
